domain/model/state/controller/state.go

```diff
@@ -2176,3 +2176,53 @@ WHERE uuid = $dbModelUUID.uuid;`,
 
 	return !invalidCredentialStatus.CredentialInvalid, nil
 }
+
+// DefaultCloudCredentialLabelForOwner returns the owner's default cloud credential label for a given
+// cloud. If user has multiple (or no) credentials for the specified cloud a NotFound error is returned as
+// we cannot determine the default credential.
+func (st *State) DefaultCloudCredentialLabelForOwner(ctx context.Context, owner user.Name, cloudName string) (string, error) {
+	db, err := st.DB(ctx)
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	var credNames []dbName
+	credQuery := `
+SELECT DISTINCT (cc.name) AS (&dbName.*)
+FROM   v_cloud_credential_attribute cc
+WHERE  cc.owner_name = $ownerAndCloudName.owner_name
+AND    cc.cloud_name = $ownerAndCloudName.cloud_name
+AND    cc.revoked = false
+AND    cc.invalid = false
+`
+	names := ownerAndCloudName{
+		OwnerName: owner.Name(),
+		CloudName: cloudName,
+	}
+	credStmt, err := st.Prepare(
+		credQuery,
+		dbName{},
+		names,
+	)
+	if err != nil {
+		return "", errors.Errorf("preparing select credentials for owner statement: %w", err)
+	}
+
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+
+		err = tx.Query(ctx, credStmt, names).GetAll(&credNames)
+		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
+			return errors.Errorf("loading cloud credentials: %w", err)
+		}
+		return nil
+	})
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	if len(credNames) != 1 {
+		return "", modelerrors.CredentialNotFound
```

Comment:
You need to add context to this with New or Errorf. It is not good enough to just return the error.